### DISCUSSION TITLE
different locales, different style and directions

### DISCUSSION THREAD

**Discussion by vahidtakro** [vinissimus/next-translate#668] [Q&A]
Hello, thanks for building this amazing package.
I am a new nextjs learner, so the question may be very basic.

How can I make it possible when user changes locale, the direction (rtl/ltr) changes too? how can I set different style files for each locale?

**Comment by aralroca** [comment]
@vahidtakro The language is in the `html` element.

![image](https://user-images.githubusercontent.com/13313058/130264657-9323ff41-6ced-4bbf-9445-a1f465034b1a.png)

You can directly use `html[lang=es]` in CSS files to make the styles affect only the Spanish language.

However, at any point, you can access the language with the `useTranslate` hook, and create your own classes wherever you want.

If you don't mind I move the issue as a discussion. All questions and answers are best asked there 😊

**Comment by rhernandog** [comment]
Hello,

I was wondering if it would be possible to add a direction property to the `useTranslation` hook. We have several languages in our app and a few of them use RTL and I don't want to run a check if the selected language is part of an array. The main reason is not that the check would be long or something like that, but the fact that at some point someone could forget to update such array. It would be great if we could add the direction in the `i18n.json` file, like this:

```json
{
  "locales": [
    {
      "lang": "en",
      "direction": "ltr"
    },
    {
      "lang": "ar",
      "direction": "rtl"
    }
  ]
}
```

**Reply by aralroca** [reply]
There is no plan for this. To use `rtl` should be specify it via CSS.

>  I don't want to run a check if the selected language is part of an array

You can modify the `i18n.json` to `i18n.js` and use `import`, and unify in your constants what you need and what next-translate needs. And is not necessary an array for the direction, you can use a set, or an object.
